html/browsers/history/the-location-interface/location-prototype-setting-goes-cross-origin-domain.sub.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>[[SetPrototypeOf]] on a Location object should not allow changing its value: cross-origin via document.domain after initially getting the object</title>
+<link rel="help" href="http://html.spec.whatwg.org/multipage/#location-setprototypeof">
+<link rel="author" title="Domenic Denicola" href="mailto:d@domenic.me">
+
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/test-setting-immutable-prototype.js"></script>
+
+<iframe src="/common/blank.html"></iframe>
+
+<script>
+"use strict";
+setup({ explicit_done: true });
+
+window.onload = () => {
+  const targetLocation = frames[0].location;
+
+  test(() => {
+    assert_not_equals(Object.getPrototypeOf(targetLocation), null);
```

Comment:
Why not store this object in a variable and try setting targetLocation's prototype to it after you set document.domain?